airflow/api/common/mark_tasks.py

```diff
@@ -361,7 +358,9 @@ def _set_dag_run_state(dag_id: str, run_id: str, state: DagRunState, session: SA
     :param state: target state
     :param session: database session
     """
-    dag_run = session.query(DagRun).filter(DagRun.dag_id == dag_id, DagRun.run_id == run_id).one()
+    dag_run = session.execute(
+        select(DagRun).where(DagRun.dag_id == dag_id, DagRun.run_id == run_id)
+    ).scalar_one()
```

Comment:
Should be equivalent to

```suggestion
    dag_run = session.scalars(
        select(DagRun).where(DagRun.dag_id == dag_id, DagRun.run_id == run_id)
    ).one()
```

but I _don’t_ think this is better. So just for reference in case anyone wonders.